app/controllers/have_read_books_controller.rb

```diff
@@ -2,15 +2,41 @@ class HaveReadBooksController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @current_month = Time.zone.now.strftime("%b")
-    @current_day = Time.zone.now.day
-    @current_year = Time.zone.now.year
     @have_read_books = current_user.completed_books.order_by(sort_params, HaveReadBook).includes(:authors, :genres, :have_read_books)&.page params[:page]
   end
 
+  def create
+    book = Book.find(have_read_params[:book_id])
+    have_read_book = HaveReadBook.new(user: current_user, book_id: have_read_params[:book_id])
+
+    if have_read_book.save
+      redirect_to book_path(book), notice: "Book successfully added to Have Read"
+    else
+      redirect_to book_path(book), alert: "Book failed to save to Have Read"
+    end
+  end
+
   def update
-    @have_read_book = HaveReadBook.find_by(user: current_user, book_id: params[:id])
-    @have_read_book.build_date(have_read_params)
+    @book = Book.find(have_read_params[:book_id])
+    @have_read_book = HaveReadBook.find_by(user: current_user, book_id: have_read_params[:book_id])
+    @have_read_book.build_date(date_params)
+    @have_read_book.assign_attributes(have_read_params)
+
+    if @have_read_book.save
+      redirect_to book_path(@book), notice: "Have Read book successfully updated"
+    else
+      redirect_to book_path(@book), alert: "Have Read book failed to update"
+    end
+  end
+
+  def create_or_update
+    have_read_book = HaveReadBook.find_by(user: current_user, book_id: have_read_params[:book_id])
```

Comment:
You can avoid loading the whole activerecord object here by calling .exists? to get a true/false